akka-cluster-tools/src/main/scala/akka/cluster/client/ClusterClient.scala

```diff
@@ -812,12 +812,18 @@ object ClusterReceptionist {
      */
     class ClientResponseTunnel(client: ActorRef, timeout: FiniteDuration) extends Actor with ActorLogging {
       context.setReceiveTimeout(timeout)
+
+      private val isAsk = client.path.elements.headOption.contains("temp")
```

Comment:
maybe `client.path.toString startsWith "/temp/$` would be a bit future-safer?
I was checking if we don't register FunctionRefs here but seems we don't... Do you think this is something to worry about or not really? Not sure if Typed needs to register temp actors hm